src/SilbinaryWolf/Components/ComponentService.php

```diff
@@ -158,13 +162,18 @@ public function createProperty($name, array $parts)
      * @param  SSViewer_Scope $scope
      * @return DBHTMLText
      */
-    public function renderComponent($name, array $props, SSViewer_Scope $scope)
+    public function renderComponent($name, array $props, SSViewer_Scope $scope = null)
```

Comment:
I'd rather the test pass in an `new SSViewer_Scope` **assuming** that when renderComponent() is called in non-test code it never passes in `null`.

Tests should align with reality as closely as possible.

Reply:
I was thinking of removing it completely because `renderComponent` never uses the variable - but can create a dummy object (I think... I can't see any references of an SSViewer_Scope object being created in the codebase so not sure if it's going to be a sane process :)

Reply:
I think the reasoning for including was that <% include %> has access to the $scope, so my thinking  was keep it around in the case that somebody/we might want to add an extension point that utilizes that $scope information and brings in some variables.

Buuuut, that would make components no longer pure. (I probably should have just killed that argument)